### DISCUSSION TITLE
Why are you keeping all my history

### DISCUSSION THREAD

**Discussion by JohnDizzle** [MicrosoftEdge/WebView2Feedback#4001] [Q&A]
I would think that utilizing your webview2 and forging an app that will track or delete history either in IndexedDb or through a simple Navigation handler capturing of the history from the webview2 that it an understanding that all is kept or lost after an instance is garbage collected.    Although MS has been keeping all my traffic / history / search / visits in a sqllite format db in my localstate folder..    Why is this?

**Comment by david-risney** [comment]
When you create a `CoreWebView2Environment` you specify (or leave null and get the default) the user data folder path which is where WebView2 will store state on the disk. WebView2 matches the Edge browser and other chromium-based browsers in this behavior and what is stored in the user data folder. If you have a WebView2 and the end user logs in to a website that stores the user's authentication information in cookies, those cookies are stored in the user data folder. The next time your app (possibly a different instance of your app) creates a `CoreWebView2Environment` using that same user data folder the state stored in the user data folder will be used again, and the end user will still be logged in to the website.

If you want to clear information from the user data folder you can use the [CoreWebView2Profile.ClearBrowsingDataAsync method](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2profile.clearbrowsingdataasync).

You can read more about the user data folder, what exactly is stored there, how to delete things from it, and more in [the User Data Folder document](https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/user-data-folder).

**Comment by JohnDizzle** [comment]
I understand all well on how everything works.  You are referring to the cache, and the cookies which is understandable.  I am referring multiple files that are database files that are stored in the ebview/default directory.    History is one of those files in the default directory which stores all your navigation history and searches.   I also understand chromium dev tools that MS has forked to methods and others that you may call independently.    Even after a clearing my cache and cookies that ONE FILE "HISTORY" still store my history and search.   Please understand that this a UWP app and a CoreWebViewProfile doesn't exist yet...
